File: js/src/Ice/browser/WSEndpointFactory.js
// **********************************************************************
//
// Copyright (c) 2003-2014 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************


var Ice = require("../Ice/ModuleRegistry").Ice;
Ice.__M.require(module, ["../Ice/Class", "../Ice/Endpoint", "../Ice/browser/WSEndpoint"]);

var WSEndpoint = Ice.WSEndpoint;

var WSEndpointFactory = Ice.Class({
    __init__:function(instance, secure)
    {
        this._instance = instance;
        this._secure = secure;
    },
    type: function()
    {
        return this._secure ? Ice.WSSEndpointType : Ice.WSEndpointType;
    },
    protocol: function()
    {
        return this._secure ? "wss" : "ws";
    },
    create: function(str, oaEndpoint)
    {
        return WSEndpoint.fromString(this._instance, this._secure, str, oaEndpoint);
    },
    read: function(s)
    {
        return WSEndpoint.fromStream(s, this._secure);
    },
    destroy: function()
    {
        this._instance = null;
    }
});
Ice.WSEndpointFactory = WSEndpointFactory;

File: java/src/Ice/src/main/java/com/zeroc/Ice/WSEndpointFactory.java
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

package com.zeroc.Ice;

final class WSEndpointFactory extends EndpointFactoryWithUnderlying {
    public WSEndpointFactory(ProtocolInstance instance, short type) {
        super(instance, type);
    }

    @Override
    public EndpointFactory cloneWithUnderlying(ProtocolInstance instance, short underlying) {
        return new WSEndpointFactory(instance, underlying);
    }

    @Override
    public EndpointI createWithUnderlying(
            EndpointI underlying, java.util.ArrayList<String> args, boolean oaEndpoint) {
        return new WSEndpoint(_instance, underlying, args);
    }

    @Override
    public EndpointI readWithUnderlying(EndpointI underlying, InputStream s) {
        return new WSEndpoint(_instance, underlying, s);
    }
}
